### ISSUE TITLE
Why does the ppx have a separate opam file? 

### ISSUE THREAD

**Issue by rikimasan** [reasonml/reason-react#827]
This ends up creating multiple copies of the same code locally which is causing issues with jest haste map finding two instances of the same thing. Found while using the metro bundler for a react native project. Seems like this could all be under one opam file and just have the ppx be reason-react.ppx especially since you don't want the ppx version to mismatch the reason-react version.

**Comment by rikimasan** [comment]
We were able to fix this for now by adding the following to our metro.config.js, however I'm not sure this is the ideal solution? I don't think metro should need to know about the opam folder so it's probably fine but I would still like some insight as to why the choice to have them separate was taken.
```
const config = {
    // ...
    resolver: {
      blacklistRE: blacklist(
        [
          /.*\/\_opam\/.*/,
        ].concat(
          workspaces.map(
            (workspacePath) =>
              `/${workspacePath.replace(/\//g, '[/\\\\]')}[/\\\\]node_modules[/\\\\]react-native[/\\\\].*/`
          ))
      ),
      // ...
    },
  };
```

**Comment by anmonteiro** [comment]
> Why does the ppx have a separate opam file?

Because it's a different package than the reason-react library. It could even be used with other libraries as long as they remain compatible with the generated PPX API.

As for the `_opam` folder, you're right that metro shouldn't need to know about it. If you're using Melange with Dune + OPAM, you'll get all the JS files in the `melange.emit` target directory -- those are all the artifacts that your bundler should care about.

**Comment by davesnx** [comment]
Hey @rikimasan, can you share a minimised example where metro reads stuff from _opam?

I would like to understand why a separate opam package causes an issue here. 

Thanks. (I will keep the issue closed since there's non issue in reason-react, but happy to move this further)

**Comment by rikimasan** [comment]
I went to reproduce this but wasn't able to do it in a more simple repo, granted the repo I tested in this time is on a much newer set of packages. 

I think this issue was mainly created in regards to the duplication of files that results from having two packages in the same repository though. Perhaps detecting shared repos is something that should be addressed on the opam side of things though since this is a supported use case for dune.
